test/e2e/keycloaks_test.go

```diff
@@ -42,6 +45,201 @@ func NewUnmanagedKeycloaksCRDTestStruct() *CRDTestStruct {
 	}
 }
 
+func NewKeycloaksSSLTestStruct() *CRDTestStruct {
+	return &CRDTestStruct{
+		prepareEnvironmentSteps: []environmentInitializationStep{
+			prepareKeycloaksSSLWithDB,
+		},
+		testSteps: map[string]deployedOperatorTestStep{
+			"keycloakSSLDBTest": {testFunction: keycloakSSLDBTest},
+		},
+	}
+}
+
+func keycloakSSLDBTest(t *testing.T, f *framework.Framework, ctx *framework.Context, namespace string) error {
+	// get the Keycloak Statefulset
+	keycloakStatefulset := v1apps.StatefulSet{}
+	err := GetNamespacedObject(f, namespace, model.KeycloakDeploymentName, &keycloakStatefulset)
+	if err != nil {
+		return err
+	}
+
+	// check pod has the env var JDBC_PARAMS
+	envExists := false
+	for _, envvar := range keycloakStatefulset.Spec.Template.Spec.Containers[0].Env {
+		if envvar.Name == "JDBC_PARAMS" && strings.Contains(envvar.Value, "sslmode") {
+			envExists = true
+			break
+		}
+	}
+	if !envExists {
+		return errors.Errorf("test Failed : Env var JDBC_PARAMS and value sslMode not found")
+	}
+
+	// check the volume to the crt exists too
+	volumeExists := false
+	for _, vol := range keycloakStatefulset.Spec.Template.Spec.Volumes {
+		if vol.Name == model.DatabaseSecretSslCert+"-vol" {
+			volumeExists = true
+			break
+		}
+	}
+	if !volumeExists {
+		return errors.Errorf("test Failed : Volume to the secret not found")
+	}
```

Comment:
I believe you don't need to use a `for` loops which is a bit cumbersome. You can do something like [this](https://stackoverflow.com/a/2050629) instead. But it's just a nitpick – I'll leave it up to you if you want to change it. ;)